### PR TITLE
APIGOV-19648 - jobs and discovery cache performance changes

### PR DESCRIPTION
**Pull request by jcollins-axway**
- Added ability to set page size when using api server paging methods
- Set page size for discovery cache to 100
- removed query filter on discovery cache and check locally for externalapiid attr
- getting api service instances in place of consumer instances
-- using cache to check for apis removed on dataplane
- only querying for new services and instances, hourly querying for all services and instances
- backoff implemented in jobs api

Note: not using the attribute on the agent resource, replaced with the timestamp query

### CODE REVIEW

**Review by vivekschauhan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Review by dfeldick — APPROVED** [review]

**Review by kflan-io — APPROVED** [review]

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -161,12 +163,27 @@ func OnAgentResourceChange(agentResourceChangeHandler ConfigChangeHandler) {
...
 	}
-	log.Tracef("registered API cache update job: %s", id)
+	log.Tracef("registered API cache update all job: %s", id)
+
+	// Start the regular update after the first interval
+	go func() {
+		newDiscoveryCacheJob := newDiscoveryCache(false)
+		newDiscoveryCacheJob.lastInstanceTime = time.Now()
+		newDiscoveryCacheJob.lastServiceTime = time.Now()
+		time.Sleep(agent.cfg.GetPollInterval())
```
Comment (sbolosan) [inline pkg/agent/agent.go:179]: Will having this configurable be an issue? Haven't checked, but do we validate this config for < 0 or non numeric

Comment (jcollins-axway) [reply pkg/agent/agent.go:179]: Has to be a positive value, > 0

Comment (jcollins-axway) [reply pkg/agent/agent.go:179]: Also the cfg library has it set as an int, so that validation is done

Comment (sbolosan) [reply pkg/agent/agent.go:179]: Too lazy to check just wanted to make sure they can't do -1

**Review by sbolosan — APPROVED** [review]
The change to page size of 100 a seby recommendation?

**Comment by jcollins-axway** [comment]
> The change to page size of 100 a seby recommendation?

Yes, we discussed that since we are limiting the fields to name,title,attributes

File: pkg/agent/agent.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/agent/agent.go:179]: We, as discussed, should still add a floor, as 1ns is valid....

**Review by Atraya21 — APPROVED** [review]

**Review by tsjohns9 — APPROVED** [review]

**Pull request merged**